pepys_timeline/static/js/src/pepys.js

```diff
@@ -6,7 +6,56 @@ const DATE_FORMATS = {
   picker: "DD/MM/YYYY",
 }
 
-let generatedCharts = false;
+const countdownNumberEl = document.getElementById('countdown-number');
+const progress = document.querySelector('#countdown-progress');
+
+const resetCountdown = () => {
+  countdownNumberEl.textContent = config.frequency_secs;
+  updateCountdownProgress(config.frequency_secs);
+}
+
+const updateCountdownProgress = (seconds) => {
+  period = config.frequency_secs;
+  const timePct = period !== 0 ? seconds/period : 0;
+  total = Math.PI * (2 * progress.r.baseVal.value);
+  progressPct = (1 - timePct) * total;
+  progress.style.strokeDashoffset = progressPct;
+}
+
+const onTimerStarted = (event) => {
+  resetCountdown();
+}
+const onTimerSecondsUpdated = (event) => {
+  const { detail } = event;
+  const { timer: t } = detail;
+  const { seconds } = t.getTimeValues();
+  countdownNumberEl.textContent = seconds;
+  updateCountdownProgress(seconds);
+}
+const onTimerTargetAchieved = () => {
+  fetchSerialsMeta();
+  timer.reset();
```

Comment:
I _think_ the correct logic is to reset the timer _then_ fetch the data.

Otherwise the update cycle is one minute _plus_ the time to retrieve the data.

(happy to discuss, good chance I have that wrong 🤣  )

Reply:
I think JS fetch is asynchronous, so it will not wait for the whole round trip time to reset the timer, but maybe we could flip the two lines just to be completely precise.